### ISSUE TITLE
GDAL error when downloading individual images (using composite_function = NULL)

### ISSUE THREAD

**Issue by jguelat** [Permian-Global-Research/rsi#29]
First I'd like to thank you a lot for this nice package! It's a real pleasure to use it!

Unfortunately I'm getting gdalwarp errors when trying to download all available images using `composite_function = NULL`. I've tried different periods and areas of interest. The error is occurring with both landsat and sentinel2. Suprinsingly there's no problem when using a composite.

For example the following code is not working for me (same code than the one you posted in https://github.com/Permian-Global-Research/rsi/issues/6).

```r
library(rsi)

aoi <- sf::st_point(c(-74.912131, 44.080410)) |> 
  sf::st_sfc() |> 
  sf::st_set_crs(4326) |> 
  sf::st_transform(3857) |> 
  sf::st_buffer(1000)

images <- get_landsat_imagery(
  aoi = aoi,
  start_date = "2022-06-01",
  end_date = "2022-08-01",
  composite_function = NULL
)
```
I'm getting the following error message:

> Error in sf::gdal_utils("warp", vrt, out, options = gdalwarp_options) : 
>   gdal_utils warp: an error occured

I'm also getting the following warnings (note that I'm always getting the first warning, even for composites. I should probably change some option in the gdalwarp_options for this one).

> 
> 1: In CPL_gdalwarp(source, destination, options, oo, doo, config_options,  :
>   GDAL Message 6: HTTP/2 not available in this build of Curl. It needs to be built against nghttp2
> 2: In CPL_gdalwarp(source, destination, options, oo, doo, config_options,  :
>   GDAL Message 1: file47cc7a1748e6.tif, band 9: Setting nodata to -123456.789000000004 on band 9, but band 1 has nodata at nan. The TIFFTAG_GDAL_NODATA only support one value per dataset. This value of -123456.789000000004 will be used for all bands on re-opening
> 3: In CPL_gdalwarp(source, destination, options, oo, doo, config_options,  :
>   GDAL Message 1: file47cc1f3b3ca4.tif, band 9: Setting nodata to -123456.789000000004 on band 9, but band 1 has nodata at nan. The TIFFTAG_GDAL_NODATA only support one value per dataset. This value of -123456.789000000004 will be used for all bands on re-opening
> 4: In CPL_gdalwarp(source, destination, options, oo, doo, config_options,  :
>   GDAL Message 1: file47cc91e63be.tif, band 9: Setting nodata to -123456.789000000004 on band 9, but band 1 has nodata at nan. The TIFFTAG_GDAL_NODATA only support one value per dataset. This value of -123456.789000000004 will be used for all bands on re-opening
> 5: In CPL_gdalwarp(source, destination, options, oo, doo, config_options,  :
>   GDAL Message 1: file47cc492258db.tif, band 9: Setting nodata to -123456.789000000004 on band 9, but band 1 has nodata at nan. The TIFFTAG_GDAL_NODATA only support one value per dataset. This value of -123456.789000000004 will be used for all bands on re-opening
> 6: In CPL_gdalwarp(source, destination, options, oo, doo, config_options,  :
>   GDAL Message 1: file47cc4af42e09.tif, band 9: Setting nodata to -123456.789000000004 on band 9, but band 1 has nodata at nan. The TIFFTAG_GDAL_NODATA only support one value per dataset. This value of -123456.789000000004 will be used for all bands on re-opening
> 7: In CPL_gdalwarp(source, destination, options, oo, doo, config_options,  :
>   GDAL Error 4: Attempt to create new tiff file `sleep_else_fraction_2022-07-27T15:44:51.989020Z.tif' failed: No error

Here's my `sessionInfo()`, and the `sf` package is linking to GDAL 3.6.2 (default CRAN version):

> R version 4.3.1 (2023-06-16 ucrt)
> Platform: x86_64-w64-mingw32/x64 (64-bit)
> Running under: Windows 10 x64 (build 19045)
> 
> Matrix products: default
> 
> 
> locale:
> [1] LC_COLLATE=German_Switzerland.utf8  LC_CTYPE=German_Switzerland.utf8    LC_MONETARY=German_Switzerland.utf8 LC_NUMERIC=C                        LC_TIME=German_Switzerland.utf8    
> 
> time zone: Europe/Zurich
> tzcode source: internal
> 
> attached base packages:
> [1] stats     graphics  utils     datasets  grDevices methods   base     
> 
> other attached packages:
> [1] rsi_0.1.0
> 
> loaded via a namespace (and not attached):
>  [1] future.apply_1.11.0   jsonlite_1.8.7        dplyr_1.1.4           compiler_4.3.1        crayon_1.5.2          tidyselect_1.2.0      Rcpp_1.0.11           parallel_4.3.1        globals_0.16.2       
> [10] R6_2.5.1              generics_0.1.3        curl_5.1.0            classInt_0.4-10       sf_1.0-14             rstac_0.9.2-5         future_1.33.0         tibble_3.2.1          units_0.8-4          
> [19] DBI_1.1.3             pillar_1.9.0          rlang_1.1.2           utf8_1.2.4            terra_1.7-55          cli_3.6.1             magrittr_2.0.3        progressr_0.14.0      class_7.3-22         
> [28] proceduralnames_0.2.2 digest_0.6.33         grid_4.3.1            lifecycle_1.0.4       vctrs_0.6.4           KernSmooth_2.23-22    proxy_0.4-27          glue_1.6.2            listenv_0.9.0        
> [37] codetools_0.2-19      parallelly_1.36.0     fansi_1.0.5           e1071_1.7-13          httr_1.4.7            tools_4.3.1           pkgconfig_2.0.3

Thanks a lot for your help!

**Comment by mikemahoney218** [comment]
Thanks for the report! I'll dig into this more tomorrow, but my suspicion is that the default file names generated are too long for Windows. Unfortunately I don't know that there's a workaround using the current released version -- I'll need to dig in a little

**Comment by jguelat** [comment]
Thanks a lot!

**Comment by mikemahoney218** [comment]
The issue is even sillier than the names being too long -- `:` is an invalid character in file names on Windows, so adding a datetime to the end of each file to differentiate them winds up making the file name invalid. Believe this is fixed in #32 and will be in `main` by the end of the day (and on CRAN relatively soon, as I need to fix a CRAN error as well)

**Comment by jguelat** [comment]
Awesome, thanks!
